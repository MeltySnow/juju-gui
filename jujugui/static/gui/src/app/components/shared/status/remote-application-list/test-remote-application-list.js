/* Copyright (C) 2018 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const StatusRemoteApplicationList = require('./remote-application-list');

describe('StatusRemoteApplicationList', () => {
  let remoteApplications;

  const renderComponent = (options = {}) => enzyme.shallow(
    <StatusRemoteApplicationList
      remoteApplications={options.remoteApplications || remoteApplications}
      statusFilter={options.statusFilter} />
  );

  beforeEach(() => {
    remoteApplications = [{
      getAttrs: sinon.stub().withArgs().returns({
        service: 'haproxy',
        status: {current: 'unknown'},
        url: 'local:admin/saas.haproxy'
      })
    }, {
      getAttrs: sinon.stub().withArgs().returns({
        service: 'mongo',
        status: {current: 'corrupting data'},
        url: 'local:admin/my.mongo'
      })
    }];
  });

  it('renders', () => {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });
});
