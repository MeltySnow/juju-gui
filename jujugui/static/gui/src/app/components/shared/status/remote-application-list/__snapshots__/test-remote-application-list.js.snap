// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusRemoteApplicationList renders 1`] = `
<StatusTable
  headers={
    Array [
      Object {
        "columnSize": 3,
        "content": "SAAS",
      },
      Object {
        "columnSize": 3,
        "content": "Status",
      },
      Object {
        "columnSize": 3,
        "content": "Store",
      },
      Object {
        "columnSize": 3,
        "content": "URL",
      },
    ]
  }
  rows={
    Array [
      Object {
        "columns": Array [
          Object {
            "columnSize": 3,
            "content": "haproxy",
          },
          Object {
            "columnSize": 3,
            "content": "unknown",
          },
          Object {
            "columnSize": 3,
            "content": "local",
          },
          Object {
            "columnSize": 3,
            "content": "admin/saas.haproxy",
          },
        ],
        "extraData": "ok",
        "key": "local:admin/saas.haproxy",
      },
      Object {
        "columns": Array [
          Object {
            "columnSize": 3,
            "content": "mongo",
          },
          Object {
            "columnSize": 3,
            "content": "corrupting data",
          },
          Object {
            "columnSize": 3,
            "content": "local",
          },
          Object {
            "columnSize": 3,
            "content": "admin/my.mongo",
          },
        ],
        "extraData": "ok",
        "key": "local:admin/my.mongo",
      },
    ]
  }
/>
`;
